### PR TITLE
Changes and additions for Zeek for Windows

### PR DESCRIPTION
**Pull request by voidbar**

### CODE REVIEW

**Comment by voidbar** [comment]
This is an initial commit. Still trying to debug compilation errors on windows

File: regex/regex.h [hunk 1]
```diff
@@ -1,119 +1,101 @@
...
- */
-
-#ifndef _REGEX_H_
-#define	_REGEX_H_
-
-#include <unistd.h>
-#include <sys/cdefs.h>
+#pragma once
+
+#include <regex>
```
Comment (timwoj) [inline regex/regex.h:3]: This is a change I'm not sure they'll accept upstream. There's historically been some issues with the performance of C++'s regex engine on some platforms, and they might prefer the hand-written version. When we do go to upstream this stuff, we should probably split this one out into a separate PR.

File: unistd/arpa/nameser.h [hunk 2]
```diff
@@ -0,0 +1,57 @@
+// arpa/nameser.h
+// Provided by brimdata on 2022/10/19
```
Comment (timwoj) [inline unistd/arpa/nameser.h:2]: Have you talked to the Brim guys about whether they're ok with providing their implementations upstream?

File: unistd/netdb.c [hunk 3]
```diff
@@ -0,0 +1,15 @@
+// netdb.c
+// Provided by brimdata on 2022/10/19
+// Copyright (c) 2019 Robin.Rowe@CinePaint.org
+// License open source MIT
+
+#include "netdb.h"
+#include <stdlib.h>
+
+struct hostent *gethostbyname2(const char *name, int af)
+	{
```
Comment (timwoj) [inline unistd/netdb.c:10]: I'm going to see about getting a clang-format configuration committed upstream so that we can reformat things to match some sort of standard. For example, this file is in Whitesmiths, but lots of others are in other formats.

File: lmdb/liblmdb/mdb.c [hunk 4]
```diff
@@ -39,40 +39,40 @@
 #define _FILE_OFFSET_BITS	64
 #endif
 #ifdef _WIN32
-#include <malloc.h>
-#include <windows.h>
-#include <wchar.h>				/* get wcscpy() */
+	#include <malloc.h>
```
Comment (timwoj) [inline lmdb/liblmdb/mdb.c]: There's a bunch of whitespace changes in this file that could go away, leaving just the actual functional change.

**Review by timwoj — CHANGES_REQUESTED** [review]
Some comments about future work, and one minor whitespace request. Also, there's an empty file called netinet/in_systm.h. Does that file need to exist?

File: unistd/unistd.h [hunk 5]
```diff
@@ -75,6 +76,8 @@ enum
 //CFUNC pid_t getpgrp(...); /* POSIX.1 version */
 CFUNC pid_t getpgrp(pid_t pid); /* BSD version */
 CFUNC int setpgrp(pid_t pid, pid_t pgid); 
+CFUNC int read(int fh, void* buf, unsigned count);
```
Comment (timwoj) [inline unistd/unistd.h:79]: Getting a bunch of these warnings from this change:

```
C:\Users\Tim\Desktop\projects\libunistd\unistd\unistd.h(79): warning C4273: 'read': inconsistent dll linkage
  C:\Program Files (x86)\Windows Kits\10\include\10.0.19041.0\ucrt\corecrt_io.h(525): note: see previous definition of 'read'
```
